server/study_groups/methods.js

```diff
@@ -20,6 +20,12 @@ Meteor.methods({
        throw new Meteor.Error('StudyGroups.methods.createNewStudyGroup.not-logged-in', 'Must be logged in to create new Study Group.');
      }
 
+     // Check study group name is unique
+     isTitleAlreadyUsed = StudyGroups.find({title: data.title, visibility: true}).count() > 0;
```

Comment:
- Please, use block scope variable, `const  isTitleAlreadyUsed`
- use `findOne` instead of `find`, `find` will scan the entire collection- where `findOne` will exit once it find the first matching record.
- mongodb queries are case sensitive so we should use regex as a workaround here, 
`{ title: /^data.title$/i, visibility: true }`